staging/src/k8s.io/pod-security-admission/policy/check_apparmor.go

```diff
@@ -0,0 +1,93 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package policy
+
+import (
+	"fmt"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/pod-security-admission/api"
+)
+
+func init() {
+	addCheck(CheckAppArmorProfile)
+}
+
+// CheckAppArmorProfile returns a baseline level check
+// that limits the value of AppArmor profiles in 1.0+
+func CheckAppArmorProfile() Check {
+	return Check{
+		ID:    "appArmorProfile",
+		Level: api.LevelBaseline,
+		Versions: []VersionedCheck{
+			{
+				MinimumVersion: api.MajorMinorVersion(1, 0),
+				CheckPod:       appArmorProfile_1_0,
+			},
+		},
+	}
+}
+
+const (
+	permissionFormat = "container.apparmor.security.beta.kubernetes.io/%s"
+)
+
+var (
+	forbiddenValues = sets.NewString(
+		"unconfined",
+	)
+)
+
+func appArmorProfile_1_0(podMetadata *metav1.ObjectMeta, podSpec *corev1.PodSpec) CheckResult {
+	forbiddenContainers := sets.NewString()
+
+	// undefined is an allowed value for 'container.apparmor.security.beta.kubernetes.io/*'
+	if len(podMetadata.Annotations) == 0 {
+		return CheckResult{Allowed: true}
+	}
+
+	checkContainer := func(containerName string) {
+		if profile := podMetadata.Annotations[fmt.Sprintf(permissionFormat, containerName)]; forbiddenValues.Has(profile) {
+			forbiddenContainers.Insert(containerName)
+		}
+	}
+
+	// check containers
+	for _, container := range podSpec.Containers {
```

Comment:
Rather than checking each container, I think it would be better to just check each annotation that matches the AppArmor prefix. It doesn't matter if there isn't an associated container, it should still fail.